src/NuGet.Core/NuGet.ProjectModel/ToolPathResolver.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    public class ToolPathResolver
+    {
+        private readonly string _packagesDirectory;
+
+        public ToolPathResolver(string packagesDirectory)
+        {
+            _packagesDirectory = packagesDirectory;
+        }
+
+        public string GetLockFilePath(string packageId, SemanticVersion version, NuGetFramework framework)
```

Comment:
Great point, this should indeed be NuGetVersion (since a tool is a normal package). Adding this extra validation for tools is confusing. I don't think this is the right place to push that effort.